### PR TITLE
[Opentelemetry] ceph and kvm parsing for logs-collector

### PR DESCRIPTION
**Pull request by Kuckkuck**
First version of ceph and kvm parsing for Opentelemetry:

ceph: containerd, journald and k8s_events are collected.
bucket and project.id are parsed

kvm: containerd, journald, openswitch*, qemu logs and k8s_events are collected

### CODE REVIEW

File: opentelemetry/chart/templates/logs-collector.yaml [hunk 1]
```diff
@@ -192,15 +223,61 @@ spec:
...
+              - set(attributes["config.parsed"], "kvm_openvswitch") where attributes["log_level"] != nil
+
+      transform/kvm_nova_agent:
+        error_mode: ignore
+        log_statements:
+          - context: log
+            conditions:
+              - resource.attributes["k8s.daemonset.name"] == "nova-hypervisor-agents-compute-kvm"
+            statements:
+              - merge_maps(attributes, ExtractGrokPatterns(body, "%{TIMESTAMP_ISO8601:logtime}%{SPACE}%{NUMBER:process.id}%{SPACE}%{WORD:log_level}%{SPACE}%{NOTSPACE:process.name}%{SPACE}\\[%{REQUEST_ID:request.id}%{SPACE}%{REQUEST_ID:request.global_id}%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}\\]%{SPACE}", true, ["REQUEST_ID=([A-Za-z0-9-]+)"]), "upsert")
```
Comment (olandr) [inline opentelemetry/chart/templates/logs-collector.yaml:279]: What is the reason for so many pairs of `%{SPACE}%{NOTSPACE}%{SPACE}%{NOTSPACE}...` (5 pairs exactly)?

File: opentelemetry/chart/templates/logs-collector.yaml [hunk 2]
```diff
@@ -56,20 +56,48 @@ spec:
...
+          - type: regex_parser
+            regex: (?P<logtime>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3})
+          - type: time_parser
+            parse_from: attributes.logtime
+            layout: '%Y-%m-%dT%H:%M:%S.%L'
+            layout_type: strptime
+          - id: file-label
+            type: add
+            field: attributes["log.type"]
+            value: "files"
```
Comment (olandr) [inline opentelemetry/chart/templates/logs-collector.yaml:100]: Line 92: Why is a named capture group being used here? Can you reference it somewhere else? 
Line 94: Is it being used there directly or is that some other `logtime`?

Comment (Kuckkuck) [reply opentelemetry/chart/templates/logs-collector.yaml:100]: The time_parser cannot access the body, one has to define a parser first and afterwards access the field: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/291159720ba75185c185e0e8ba1cceb6a5453d52/receiver/filelogreceiver

File: opentelemetry/chart/templates/logs-collector.yaml (hunk 1 above)
Comment (Kuckkuck) [reply opentelemetry/chart/templates/logs-collector.yaml]: I have shorten the parsing.

File: opentelemetry/chart/templates/logs-collector.yaml (hunk 2 above)
Comment (olandr) [reply opentelemetry/chart/templates/logs-collector.yaml:100]: Ah I see!

**Review by timojohlo — APPROVED** [review]
lgtm

**Pull request merged**